options/ansi/generic/time-stubs.cpp

```diff
@@ -1089,7 +1092,155 @@ time_t timelocal(struct tm *) {
 	__builtin_unreachable();
 }
 
-time_t timegm(struct tm *) {
-	__ensure(!"Not implemented");
-	__builtin_unreachable();
+// The following static functions and timegm implementation were taken from sortix.
+static const int DAYS_JANUARY = 31;
+static const int DAYS_FEBRUARY = 28;
+static const int DAYS_MARCH = 31;
+static const int DAYS_APRIL = 30;
+static const int DAYS_MAY = 31;
+static const int DAYS_JUNE = 30;
+static const int DAYS_JULY = 31;
+static const int DAYS_AUGUST = 31;
+static const int DAYS_SEPTEMBER = 30;
+static const int DAYS_OCTOBER = 31;
+static const int DAYS_NOVEMBER = 30;
+static const int DAYS_DECEMBER = 31;
```

Comment:
might it be preferable to put these in an anonymous namespace rather than static, use constexpr instead of const, and update these to match style elsewhere (camelCase?)